server/datastore/mysql/software.go

```diff
@@ -2251,459 +2252,689 @@ func (ds *Datastore) ListCVEs(ctx context.Context, maxAge time.Duration) ([]flee
+	VPPAppSelfService  *bool      `db:"vpp_app_self_service"`
+	VPPAppAdamID       *string    `db:"vpp_app_adam_id"`
+	VPPAppVersion      *string    `db:"vpp_app_version"`
+	VPPAppPlatform     *string    `db:"vpp_app_platform"`
+	VPPAppIconURL      *string    `db:"vpp_app_icon_url"`
+
+	VulnerabilitiesList   *string `db:"vulnerabilities_list"`
+	SoftwareIDList        *string `db:"software_id_list"`
+	SoftwareSourceList    *string `db:"software_source_list"`
+	VersionList           *string `db:"version_list"`
+	BundleIdentifierList  *string `db:"bundle_identifier_list"`
+	VPPAppSelfServiceList *string `db:"vpp_app_self_service_list"`
+	VPPAppAdamIDList      *string `db:"vpp_app_adam_id_list"`
+	VPPAppVersionList     *string `db:"vpp_app_version_list"`
+	VPPAppPlatformList    *string `db:"vpp_app_platform_list"`
+	VPPAppIconUrlList     *string `db:"vpp_app_icon_url_list"`
+}
+
+func hostInstalledSoftware(ds *Datastore, ctx context.Context, hostID uint) ([]*hostSoftware, error) {
+	installedSoftwareStmt := `
+		SELECT
+			software_titles.id AS id,
+			host_software.software_id AS software_id,
+			host_software.last_opened_at,
+			NULL AS status
+		FROM 
+			host_software
+		INNER JOIN
+			software ON host_software.software_id = software.id
+		INNER JOIN
+			software_titles ON software.title_id = software_titles.id
+		WHERE
+			host_software.host_id = :host_id
+	`
+	installedSoftwareStmt, args, err := sqlx.Named(installedSoftwareStmt, map[string]any{
+		"host_id": hostID,
+	})
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "build named query for host installed software")
 	}
 
-	var onlySelfServiceClause string
-	if opts.SelfServiceOnly {
-		onlySelfServiceClause = ` AND ( si.self_service = 1 OR ( vat.self_service = 1 AND :is_mdm_enrolled ) ) `
+	installedSoftwareStmt = ds.reader(ctx).Rebind(installedSoftwareStmt)
+	var hostInstalledSoftware []*hostSoftware
+	err = sqlx.SelectContext(ctx, ds.reader(ctx), &hostInstalledSoftware, installedSoftwareStmt, args...)
+	if err != nil {
+		return nil, err
 	}
 
-	var excludeVPPAppsClause string
-	if !opts.IsMDMEnrolled {
-		excludeVPPAppsClause = ` AND vat.id IS NULL `
+	return hostInstalledSoftware, nil
+}
+
+func hostSoftwareInstalls(ds *Datastore, ctx context.Context, hostID uint) ([]*hostSoftware, error) {
+	softwareInstallsStmt := `
+        WITH upcoming_software_install AS (
+            SELECT
+                ua.execution_id AS last_install_install_uuid,
+                ua.created_at AS last_install_installed_at,
+                siua.software_installer_id AS installer_id,
+                'pending_install' AS status
+            FROM
+                upcoming_activities ua
+            INNER JOIN
+                software_install_upcoming_activities siua ON ua.id = siua.upcoming_activity_id
+            LEFT JOIN (
+                upcoming_activities ua2
+                INNER JOIN software_install_upcoming_activities siua2 ON ua2.id = siua2.upcoming_activity_id
```

Comment:
Probably worth commenting to explain that the ua2/siua2 join is to make sure the highest priority/earliest install attempt is the one shown. If it's easy to revise aliases, may be worthwhile but comment is probably good enough since the convention being used here is a self-join looking for a null PK because that's how you make sure you have the latest value.

Same thing on hsi2.